### PR TITLE
Add helm release component definition as part of fluxcd Intializer

### LINKED ISSUE #1603: [Feature] Revisit Helm based component
Ref: #864 

Today the Helm based component need to add a `ComponentDefinition` for every chart, this is still verbose.

Should we consider this format?

```yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: myapp
  namespace: default
spec:
  components:
    - name: demo 
      type: helm 
      properties: 
        chart: "podinfo"
        version: "5.1.4" # omit to latest
        repository: "http://oam.dev/catalog/"
        values:
          image:
            tag: "5.1.2"
```
```yaml
apiVersion: core.oam.dev/v1beta1
kind: ComponentDefinition
metadata:
  name: helm
spec:
  schematic:
    cue:
      template: |
        output: {
            apiVersion: "helm.toolkit.fluxcd.io/v2beta1"
            kind: "HelmRelease"
            ...
        }
        // an extra template
        outputs: repo: {
            apiVersion: "source.toolkit.fluxcd.io/v1beta1"
            kind: "HelmRepository"
            ...
        }
        parameter: {
            chart: string
            version?: string
            repository: string
            values?: [...#Values]
        }
```
The drawback is we lose a way to define workload type (which is important). A solution is we could have components such as: helm, helm-worker, helm-task (similar to https://github.com/oam-dev/spec/blob/v0.1.0/3.component_model.md#core-workload-types)

Similar pattern also applies to Kustomize support.

/cc @wonderflow

### PR DESCRIPTION
**Pull request by chivalryq**
**What this PR does / why we need it**:

1. add `helm` as an available componentDefinition as part of fluxcd addon. either you enable an addon or apply `vela-template/addons/auto-gen/fluxcd.yaml`, you could use that.
2. refactor other addon depending on fluxcd to use `helm` type instead of `raw` type.
3. align all init namespace to vela-system(make sure depend ref)

for addon generate script ⬇️

3. move componentDef into init and remove it from autogen file
4. remove ns object in auto-gen file and remove it from autogen file(since all initializer are in vela-system)
 
**Which issue(s) this PR fixes**:

ref #1760 

fix #1603

**Special notes for your reviewer**:

### CODE REVIEW

File: vela-templates/addons/fluxcd/definitions/helm-release.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+apiVersion: core.oam.dev/v1beta1
+kind: ComponentDefinition
+metadata:
+  name: helm-release
```
Comment (wonderflow) [inline vela-templates/addons/fluxcd/definitions/helm-release.yaml:4]: 就叫helm好了

```suggestion
  name: helm
```

File: vela-templates/addons/fluxcd/definitions/helm-release.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        					namespace: context.namespace
+        				}
+        				interval: parameter.pull_interval
+        			}
+        		}
+        	}
+        }
+
+        parameter: {
+        	source_type: "GitRepo" | "HelmRepo"
```
Comment (wonderflow) [inline vela-templates/addons/fluxcd/definitions/helm-release.yaml:65]:
```suggestion
        	repoType: "git" | "helm"
```

File: vela-templates/addons/fluxcd/definitions/helm-release.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        	//+usage=The Git or Helm repository URL, accept HTTP/S or SSH address as git url.
+        	repo_url: string
```
Comment (wonderflow) [inline vela-templates/addons/fluxcd/definitions/helm-release.yaml:67]: Definition的命名整体检查一遍，统一用过 驼峰，不能出现不统一的风格。

File: vela-templates/addons/fluxcd/definitions/helm-release.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        	//+usage=The interval at which to check for repository and relese updates.
+        	pull_interval: *"5m" | string
+        	//+usage=1.The relative path to helm chart for git source. 2. chart name for helm resource
+        	chart: string
+        	if source_type == "GitRepo" {
+        		//+usage=The Git reference to checkout and monitor for changes, defaults to master branch.
+        		branch: *"master" | string
+        	}
```
Comment (wonderflow) [inline vela-templates/addons/fluxcd/definitions/helm-release.yaml:75]:
```suggestion
        		//+usage=The Git reference to checkout and monitor for changes, defaults to master branch.
        		branch:? *"master" | string
```

作为缺省参数即可，不要在 parameter里面加`if`

Comment (chivalryq) [reply vela-templates/addons/fluxcd/definitions/helm-release.yaml]: done

File: vela-templates/addons/fluxcd/definitions/helm-release.yaml (hunk 3 above)
Comment (chivalryq) [reply vela-templates/addons/fluxcd/definitions/helm-release.yaml]: done

File: vela-templates/addons/fluxcd/definitions/helm-release.yaml (hunk 2 above)
Comment (chivalryq) [reply vela-templates/addons/fluxcd/definitions/helm-release.yaml]: done

File: charts/vela-core/templates/addons/keda.yaml [hunk 5]
```diff
@@ -7,43 +7,17 @@ data:
...
-              metadata:
-                name: kedacore
-                namespace: keda
-              spec:
-                interval: 1m
-                url: https://kedacore.github.io/charts
-            type: raw
+              chart: keda
+              repo_url: https://kedacore.github.io/charts
+              source_type: HelmRepo
```
Comment (wonderflow) [inline charts/vela-core/templates/addons/keda.yaml:19]: 命名问题

**Review by wonderflow — APPROVED** [review]
LGTM

**Pull request merged**